README.md

```diff
@@ -1,6 +1,48 @@
+
+The Job Manager aspires to bring ease and efficiency to developing and debugging workflows while seamlessly scaling to production operations management.
+
+## Key Features
+* Sits on top of either [Cromwell](https://github.com/broadinstitute/cromwell) or [dsub](https://github.com/googlegenomics/dsub)
+* Service provider interface can be extended to support other engines
+* Rich, search capabilities across current and historic workflows
+* Aborting workflows
+* Clean, intuitive UX based on material design principles
+
+### Future Features
+* Dynamic grouping, filtering, and drill-down
+* Re-launching workflows
+* Simplified troubleshooting of failed workflows
+
+## Roadmap
+
+The current code is a work in progress towards an alpha release and as such has started with core features: connecting to both backends, visualizing workflow and task status and metadata, quick access to log files, and simple filtering.
+
+The near-term roadmap includes improvements to failure troubleshooting, creating a robust dashboard for grouping jobs and seeing status overviews, and improving handling of widely scattered workflows.
+
+We envision a product with user-customizable views of jobs running, insights into workflow compute cost, the ability to re-launch jobs, and the potential to make custom reports about the jobs that have been run.
+
+## Architecture Overview
+
+The Job Manager [defines an API](api/jobs.yaml) via OpenAPI. An Angular2 UI is provided over the autogenerated Typescript bindings for this API. The UI is configurable at compilation time to support various deployment environments (see [environment.ts](ui/src/environments/environment.ts)), including auth, cloud projects, and label columns.
+
+The UI must be deployed along with a backend implementation of the API; two such implementations are provided here:
+
+### Cromwell
+Monitors jobs launched by the [Cromwell workflow engine](https://github.com/broadinstitute/cromwell). The Python Flask wrapper was created using Swagger Codegen and can be configured to pull data from a specific Cromwell instance. The Job Manager currently supports Cromwell version 29.
+
+### dsub
+
+Monitors jobs that were launched via the [dsub](https://github.com/googlegenomics/dsub) CLI. Thin stateless wrapper around the dsub Python library. Authorization is required for deploying the UI, which is used to communicate with the [Google Genomics Pipelines API](https://cloud.google.com/genomics/pipelines). The wrapper itself is implemented in Python Flask using Swagger codegen models.
+
+Note that a “task” in dsub nomenclature corresponds to a Job Manager API’s “job”. A Dockerfile is provided which serves for production deployment using gunicorn.
```

Comment:
Now that I'm rereading what I wrote, perhaps move the second sentence up to the end of the previous paragraph.